mod/pages/start.php

```diff
@@ -419,6 +421,39 @@ function pages_register_db_seeds(\Elgg\Hook $hook) {
 	return $seeds;
 }
 
+/**
+ * Move child pages when parent page is deleted
+ *
+ * @elgg_event delete object
+ *
+ * @param \Elgg\Event $event Event
+ * @return void
+ */
+function pages_delete_handler(\Elgg\Event $event) {
```

Comment:
why not in the class `->delete()`??

Reply:
This way someone can stop the delete

Reply:
the action in this handler should happen on a succesful delete... having it in an event handler could cause issues if someone stops delete after this event...

Reply:
Delete event is a mess, and there is no way to deal with entity-dependent clean up. I am leaving it in a handler until there is unified way of doing this. I don't think there is a delete:after event

Reply:
> I don't think there is a delete:after event

yes there is https://github.com/Elgg/Elgg/blob/master/engine/classes/Elgg/Database/EntityTable.php#L683

Reply:
if you move it to delete:after it makes more sense.. can merge that

Reply:
It wouldn't work, because metadata would have been deleted by then. Moved to the delete method

Reply:
perfect